lib/transaction_isolation/active_record/errors.rb

```diff
@@ -6,5 +6,11 @@ module ActiveRecord
   # Serialization conflicts happen when db engine is using multi-version concurrency control.
   # Often db engines combine both approaches and thus generate both types of errors.
 
-  class TransactionIsolationConflict < ::ActiveRecord::WrappedDatabaseException; end
+  class TransactionIsolationConflict < ::ActiveRecord::WrappedDatabaseException
+    if ActiveSupport::VERSION::MAJOR == 3
+      def initialize message, original_exception = nil
+        super(message, original_exception)
+      end
```

Comment:
that does not make much sense, its default ruby to call the inherited initializer
and rails 3 is ancient

Reply:
This PR is 2 years old, not sure why you're commenting on it now..feel free to close it.

At the time a project I was working with was still on Rails 3.  Your gemspec still lists compatibility with ActiveRecord 3 (`s.add_runtime_dependency "activerecord", ">= 3.0.11"`) but the gem doesn't work with it because ActiveRecord::WrappedDatabaseException in AR 3 has 2 params - not 1 required and one optional like in AR4.

Here's AR 3's WrappedDatabaseException: https://github.com/rails/rails/blob/e17e25cd23e8abd45b1706463dd57c90fa6dcb7c/activerecord/lib/active_record/errors.rb#L67

Here's AR4's (which inherits from statement invalid - note the optional arg): https://github.com/rails/rails/blob/9be9597e510d185ca7964d0a05b4ea2a7f2d50d1/activerecord/lib/active_record/errors.rb#L63

You're only passing a single param when you use TransactionIsolationConflict internally, thus it raises an error when running with AR 3.

Another way to resolve would be to just change dependency in the gemspec to >= AR 4.

Reply:
interesting! thanks for the explanation